tasks/image-segmentation-yolov8/src/dataset.py

```diff
@@ -0,0 +1,117 @@
+from pathlib import Path
+from typing import Optional
+
+import logging
+import random
+import json
+import yaml
+
+from coretex import ImageDataset, ImageSample, CoretexImageAnnotation, ImageDatasetClasses
+
+
+def isValidationSplitValid(validationSplit: float, datasetSize: int) -> bool:
+    if not 0 <= validationSplit < 1:
+        logging.error(f">> [ObjectDetection] validationSplit parameter ({validationSplit}) must be between 0 and 1")
+        return False
+
+    minSamplesForSplit = int(1 / min(validationSplit, 1 - validationSplit))
+    if datasetSize < minSamplesForSplit:
+        logging.error(f">> [ObjectDetection] Dataset is too small ({datasetSize}) for validationSplit parameter ({validationSplit}). Minimum number of samples is {minSamplesForSplit}")
+        return False
+
+    return True
+
+
+def createYamlFile(
+    datasetPath: Path,
+    trainDatasetPath: Path,
+    validDatasetPath: Path,
+    classes: ImageDatasetClasses,
+    location: Path
+) -> None:
+
+    classesIds: dict[int, str] = {}
+    for index, clazz in enumerate(classes):
+        classesIds[index] = clazz.label
+
+    data = {
+        "path": str(datasetPath.absolute()),
+        "train": str(trainDatasetPath.relative_to(datasetPath)),
+        "val": str(validDatasetPath.relative_to(datasetPath)),
+        "names": classesIds
+    }
+
+    with open(location, "w") as yamlFIle:
+        yaml.dump(data, yamlFIle, default_flow_style = False, sort_keys = False)
+
+
+def loadAnnotation(sample: ImageSample) -> Optional[CoretexImageAnnotation]:
+    if not sample.annotationPath.exists():
+        return None
+
+    with sample.annotationPath.open("r") as file:
+        return CoretexImageAnnotation.decode(json.load(file))
+
+
+def addImagePathForYaml(sample: ImageSample, imagesPath: Path) -> None:
+    sample.unzip()
+    sample.imagePath.link_to(imagesPath / f"{sample.id}{sample.imagePath.suffix}")
+
+
+def addAnnotation(sample: ImageSample, classes: ImageDatasetClasses, location: Path) -> None:
+    sample.unzip()
+
+    annotation = loadAnnotation(sample)
+    if annotation is not None:
+        normalizedAnnotations: list[tuple[int, list[float]]] = []
+
+        for instance in annotation.instances:
+            labelId = classes.labelIdForClassId(instance.classId)
+
+            if labelId is None:
+                continue
+
+            for segmentation in instance.segmentations:
+                w = annotation.width
+                h = annotation.height
+                normalizedSegmentation = [num / w if idx % 2 == 0 else num / h for idx, num in enumerate(segmentation)]
+
+                normalizedAnnotation = labelId, normalizedSegmentation
+
+                normalizedAnnotations.append(normalizedAnnotation)
```

Comment:
There is no need to create `normalizedAnnotation` variable as it is only appended to the list. Append the values directly.